sites/geohub/src/routes/api/stac/[id]/[collection]/[product_id]/+server.ts

```diff
@@ -0,0 +1,116 @@
+import { error, type RequestHandler } from '@sveltejs/kit';
+import { isSuperuser, getProductDetails, deleteProduct } from '$lib/server/helpers';
+import DatabaseManager from '$lib/server/DatabaseManager';
+
+export const GET: RequestHandler = async ({ locals, params }) => {
+	// get product details
+	/**
+	 * get product details
+	 * Needs to be logged in to get product details
+	 */
+	const session = await locals.auth();
+	if (!session) {
+		error(403, { message: 'Permission error' });
+	}
+
+	const productDetails = await getProductDetails(params.id, params.collection, params.product_id);
+	const expression = productDetails?.expression;
+	const assets = productDetails?.assets;
+	const asset_index_mapping = assets.map((asset: string, index: number) => {
+		return {
+			[`asset${index + 1}`]: asset
+		};
+	});
+
+	productDetails.expression = replaceTextWithMapping(expression, asset_index_mapping);
+
+	if (!productDetails) {
+		error(404, { message: 'Not found' });
+	}
+	return new Response(JSON.stringify(productDetails));
+};
+
+export const POST: RequestHandler = async ({ locals, params, request }) => {
+	/**
+	 * register a new product for the available stac collection
+	 * Needs to be superuser to register product
+	 */
+	// register a new product to available stac collection
+	const session = await locals.auth();
+
+	if (!session) {
+		error(403, { message: 'Permission error' });
+	}
+
+	const user_email = session?.user.email;
+
+	let is_superuser = false;
+	if (user_email) {
+		is_superuser = await isSuperuser(user_email);
+	}
+	if (!is_superuser) {
+		error(403, { message: 'Permission error' });
+	}
+	const requestBody = await request.json();
+	const assets = requestBody.assets;
+	const stac_id = params.id;
+	const collection_id = params.collection;
+	const product_id = params.product_id;
+
+	const dbm = new DatabaseManager();
+	const client = await dbm.start();
+
+	const query = {
+		text: `INSERT INTO geohub.stac_collection_product (stac_id, collection_id, product_id, assets) VALUES ($1, $2, $3, $4)`,
+		values: [stac_id, collection_id, product_id, assets]
+	};
+
+	await client
+		.query(query)
+		.then(() => {
+			client.release();
+		})
+		.catch((e) => {
+			client.release();
+			error(500, { message: e.message });
+		});
```

Comment:
use `dbm.transactionStart` and `dbm.transactionEnd` instead. call rollback in catch. Use finally to call `transactionEnd`